### ISSUE TITLE
panic: Value missing for hash

### ISSUE THREAD

**Issue by ligi** [idena-network/idena-go#893]
I just had this crash:

```
panic: Value missing for hash 89a7948a07695ed8f6d35f776ce0e348602739fe3085663bfd6bbecb35cadb9a corresponding to nodeKey 6e89a7948a07695ed8f6d35f776ce0e348602739fe3085663bfd6bbecb35cadb9a

goroutine 7995 [running]:
github.com/cosmos/iavl.(*nodeDB).GetNode(0xc00a6e8f60, 0xc010873320, 0x20, 0x20, 0x0)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/nodedb.go:91 +0x6f7
github.com/cosmos/iavl.(*Node).getLeftNode(...)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:423
github.com/cosmos/iavl.(*Node).get(0xc0108b95e0, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0x1, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x21a
github.com/cosmos/iavl.(*Node).get(0xc0108b9540, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc0108b94a0, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc00382f360, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc006fa9a40, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0x1, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc0036a5720, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0x1, 0xc0101d40c0, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc00a21afa0, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x73, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*Node).get(0xc00a21af00, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0xffffffffffffffff, 0x2f3f468, 0xc00a6e8f60)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc00a21a500, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0x1, 0xc017b796c8, 0x18)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:173 +0x1af
github.com/cosmos/iavl.(*Node).get(0xc009baf2c0, 0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0xc017b796c8, 0x9aa78e120d587054, 0xc000088000, 0xc001ad5800)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/node.go:176 +0x273
github.com/cosmos/iavl.(*ImmutableTree).Get(0xc003937e90, 0xc017b796c8, 0x15, 0x18, 0x15, 0x18, 0x1, 0x18)
        /home/runner/go/pkg/mod/github.com/idena-network/iavl@v0.12.3-0.20211223100228-a33b117aa31e/immutable_tree.go:153 +0x5a
github.com/idena-network/idena-go/core/state.(*MutableTree).Get(0xc0103f89c0, 0xc017b796c8, 0x15, 0x18, 0x0, 0x0, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/state/tree.go:118 +0xb4
github.com/idena-network/idena-go/core/state.(*StateDB).getStateAccount(0xc00a8046e0, 0x739632743d3979d7, 0xa8903b9a3e994032, 0x9c000000db8e22ce, 0x10)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:658 +0x207
github.com/idena-network/idena-go/core/state.(*StateDB).GetOrNewAccountObject(0xc00a8046e0, 0x739632743d3979d7, 0xa8903b9a3e994032, 0xdb8e22ce, 0xffac4374ce945d00)
        /home/runner/work/idena-go/idena-go/core/state/statedb.go:844 +0x3f
github.com/idena-network/idena-go/core/state.(*NonceCache).getAccount(0xc0056c4480, 0x739632743d3979d7, 0xa8903b9a3e994032, 0x4fdb8e22ce, 0xc00e2e3c60)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:76 +0x37b
github.com/idena-network/idena-go/core/state.(*NonceCache).UnsafeSetNonce(...)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:67
github.com/idena-network/idena-go/core/state.(*NonceCache).SetNonce(0xc0056c4480, 0x739632743d3979d7, 0xa8903b9a3e994032, 0x4fdb8e22ce, 0xd)
        /home/runner/work/idena-go/idena-go/core/state/nonce_cache.go:63 +0xa5
github.com/idena-network/idena-go/core/mempool.(*TxPool).put(0xc0028b4960, 0xc016dcdd40, 0xc01aedb5c0, 0x3)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:426 +0x256
github.com/idena-network/idena-go/core/mempool.(*TxPool).add(0xc0028b4960, 0xc016dcdd40, 0xc01aedb5c0, 0xdb8e2200, 0x3, 0x0, 0xc001ad5888)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:351 +0x565
github.com/idena-network/idena-go/core/mempool.(*TxPool).AddExternalTxs(0xc0028b4960, 0x3, 0xc00e2e3fb8, 0x1, 0x1, 0x0, 0x0)
        /home/runner/work/idena-go/idena-go/core/mempool/txpool.go:280 +0x2e5
github.com/idena-network/idena-go/core/mempool.(*AsyncTxPool).loop(0xc002880a30)
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:73 +0x157
created by github.com/idena-network/idena-go/core/mempool.NewAsyncTxPool
        /home/runner/work/idena-go/idena-go/core/mempool/async_txpool.go:27 +0x95
```

I updated to 0.28.3 30min before the validation session on 1.1.2k22 and it crashed in the phase of reaching consensus

**Comment by ubiubi18** [comment]
Mine too

**Comment by sidenaio** [comment]
Fixed in #899
